### PR TITLE
Improve error message when python 3 not available - Fix for  #1385

### PR DESCRIPTION
**Pull request by lkarthee**
https://github.com/WebAssembly/wabt/issues/1385

In this PR, I have made python optional in CMakeLists.txt . I have tested locally by bumping the required version to 3.8 which is not existing on my Mac.

I have noticed that the below tests are run from python file - test/run-c-api-examples.py. 
-  c_api_example(callback)
-   c_api_example(finalize)
-   c_api_example(global)
-   c_api_example(hello)
-   c_api_example(hostref)
-   c_api_example(multi)
-   c_api_example(memory)
-   c_api_example(reflect)
-   c_api_example(serialize)
-   c_api_example(start)
-   c_api_example(table)
-   c_api_example(trap)

Do we need to port run-c-api-examples.py to gtest file or cpp file?

### CODE REVIEW

**Comment by sbc100** [comment]
Can't you already do this by building with `-DBUILD_TESTS=OFF`?  

wabt's primary test runner ./test/run-tests.py is also written in python.  If you don't have python the only possible tests you could run are the gtest unittests and i'm not sure it worth the extra option to allow just those tests to build built.

**Comment by sbc100** [comment]
Also, if you do want to run tests, isn't it easy install python3 on mac?

**Comment by lkarthee** [comment]
Installing python3 is easy. I use pyenv to manage python versions. I wanted to contribute to this project, so I picked up #1385 with `good-first-bug` tag.

To test this PR I used a python version 3.8 which is not installed in my MacBook as I don't have Windows as mentioned in #1385. 

All 5 CI checks have python3 enabled and run python tests - they won't test the scenario where python3 is not installed.

**Comment by sbc100** [comment]
Would you mind instead improving the error message, e.g. "python3 is required to for wabt testing,  please either install it or build with -DBUILD_TESTS=OFF".

**Comment by lkarthee** [comment]
I will modify the PR accordingly.

**Comment by lkarthee** [comment]
I have made required changes. See sample error message (ignore exact versions like 3.7.3 and 3.8 in the error message).

```
-- Could NOT find PythonInterp: Found unsuitable version "3.7.3", but required is at least "3.8" (found /usr/bin/python3)
CMake Error at CMakeLists.txt:587 (message):
  Python 3 is required for wabt testing.  Please install it.

  If you want to proceed without installing python 3, either add -no-tests to
  make target (example: make clang-debug-no-tests) or run cmake with
  -DBUILD_TESTS=OFF.


-- Configuring incomplete, errors occurred!
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -521,67 +521,71 @@ if (NOT EMSCRIPTEN)
...
-    c_api_example(trap)
-    if (NOT WIN32)
-      # depends on pthreads
-      set(THREADS_PREFER_PTHREAD_FLAG ON)
-      find_package(Threads REQUIRED)
-      c_api_example(threads)
-    endif ()
+    else()
+      message(FATAL_ERROR "Python 3 is required for wabt testing. Please install it. \nIf you want to proceed without installing python 3, either add -no-tests to make target (example: make clang-debug-no-tests) or run cmake with -DBUILD_TESTS=OFF.")
+    endif(PYTHONINTERP_FOUND)
```
Comment (sbc100) [inline CMakeLists.txt:588]: Seem like you could just put this error in an if an "if (NOT PYTHONINTERP_FOUND) .. endif()" above that the rest of the file would be left unchanged?

Comment (lkarthee) [reply CMakeLists.txt]: Done

**Comment by aardappel** [comment]
Would be better if instead it displayed that warning and then turned tests off for you, rather than failing.
Tests are mostly for contributors to WABT, and should not concern people simply trying to build the binaries.

**Comment by binji** [comment]
I tend to agree with @aardappel here. @sbc100, does that seem OK to you?

**Comment by sbc100** [comment]
Find with me.   We can set BUILD_TESTS to false is python is not found.

**Comment by sbc100** [comment]
Or should we even just have BUILD_TESTS default to OFF?

**Comment by binji** [comment]
I like having it default to on, personally.

**Comment by lkarthee** [comment]
Should I revert the initial commit of this PR? 
-  BUILD_TESTS default is ON (No change in existing code)
-  skip tests when python3  >= 3.5.

**Comment by sbc100** [comment]
I think the conclusion is:

1. BUILD_TESTS defaults to ON
2. BUILD_TESTS gets forced to OFF if python is now found

**Comment by lkarthee** [comment]
> - skip tests when python3 >= 3.5.

Got it - Oops typo, it should have been < sign.

File: CMakeLists.txt [hunk 2]
```diff
@@ -464,6 +464,13 @@ if (NOT EMSCRIPTEN)
     )
   endif ()
 
+  # Python 3.5 is the version shipped in Ubuntu Xenial
+  find_package(PythonInterp 3.5)
+  if(NOT PYTHONINTERP_FOUND)
```
Comment (sbc100) [inline CMakeLists.txt:469]: how about AND BUILD_TESTS == ON there?

Otherwise lgtm!

**Comment by lkarthee** [comment]
Cmake prints a message when suitable version is not found. This will appear irrespective of BUILD_TEST is ON or OFF.

When BUILD_TESTS=OFF and python3 not found
```
.......
-- Check size of size_t - done
-- Could NOT find PythonInterp: Found unsuitable version "3.7.3", but required is at least "3.8" (found /usr/bin/python3)
-- Looking for pthread.h
.......
```

When BUILD_TESTS=ON and python3 not found
```
.......
-- Check size of size_t - done
-- Could NOT find PythonInterp: Found unsuitable version "3.7.3", but required is at least "3.8" (found /usr/bin/python3)
CMake Warning at CMakeLists.txt:471 (message):
  Skipping tests.  Python 3 is required for wabt testing.  Please install
  python3 to run tests.


-- Looking for pthread.h
.......
```

File: CMakeLists.txt (hunk 2 above)
Comment (lkarthee) [reply CMakeLists.txt]: added this check.

**Pull request merged**